lib/ansible/executor/task_executor.py

```diff
@@ -486,7 +486,10 @@ def _execute(self, variables=None):
             try:
                 result = self._handler.run(task_vars=variables)
             except AnsibleConnectionFailure as e:
-                return dict(unreachable=True, msg=to_text(e))
+                failure_dict = dict(unreachable=True,
+                                    msg=to_text(e),
+                                    _ansible_connection_exception=e)
```

Comment:
Updated to extract any data bundled in the exception into the TaskResult before returning it from the worker.